server/config.go

```diff
@@ -179,13 +183,22 @@ func ReadConfig(ctx *cli.Context) Config {
 			Timeout:          ctx.Duration(S3TimeoutFlagName),
 		},
 		ClientConfig: clients.EigenDAClientConfig{
-			RPC:                          ctx.String(EigenDADisperserRPCFlagName),
-			StatusQueryRetryInterval:     ctx.Duration(StatusQueryRetryIntervalFlagName),
-			StatusQueryTimeout:           ctx.Duration(StatusQueryTimeoutFlagName),
-			DisableTLS:                   ctx.Bool(DisableTLSFlagName),
-			ResponseTimeout:              ctx.Duration(ResponseTimeoutFlagName),
-			CustomQuorumIDs:              ctx.UintSlice(CustomQuorumIDsFlagName),
-			SignerPrivateKeyHex:          ctx.String(SignerPrivateKeyHexFlagName),
+			RPC:                      ctx.String(EigenDADisperserRPCFlagName),
+			StatusQueryRetryInterval: ctx.Duration(StatusQueryRetryIntervalFlagName),
+			StatusQueryTimeout:       ctx.Duration(StatusQueryTimeoutFlagName),
+			DisableTLS:               ctx.Bool(DisableTLSFlagName),
+			ResponseTimeout:          ctx.Duration(ResponseTimeoutFlagName),
+			CustomQuorumIDs:          ctx.UintSlice(CustomQuorumIDsFlagName),
+			SignerPrivateKeyHex: func() string {
```

Comment:
actually, if someone forgot to set the key, and does not enable read only. Then the server can send unauthenticated data

Reply:
Yes, I realized this as well, I was able to request for dispersal even with random private keys. This was confusing because I thought disperser had authentication checks with ratelimits

Reply:
disperser has a free tier for testnet. but it does not have a free tier for mainnet.

Reply:
the server won't be able to send unauthenticated data on the mainnet due to rate-limits, right? what's the desirable behavior in this case? (I'm going to update this random generation to use a constant address like all zeros)